data/blog/2025/2025-02-13-mezzio101-creating-pages-and-modules.md

```diff
@@ -0,0 +1,198 @@
+draft: false
+public: true
+created: '2025-02-13T11:00:00-01:00'
+updated: '2025-02-13T11:00:00-01:00'
+tags:
+    - mezzio
+    - skeleton
+    - tutorial
+    - content-management
+    - cli
+---
+
+In the previous article we used the Mezzio Skeleton Installer to build a very basic application.
+Let's explore what we can do with it right away.
+
+The current components allow us to build a presentation site with static pages.
+It may not be much to speak of yet, but we can already create new pages and organize them in multiple modules.
+
+<!--- EXTENDED -->
+
+### Creating a new page
+
+From the beginning, the Mezzio Skeleton Installer creates a home page that lists the components we chose during installation.
+Feel free to edit that page however you see fit.
+Don't worry about any dynamic twig elements for now.
+
+What if you want a new page, like an 'About Us' page?
+Let's go through the steps of creating a new page.
+All of these steps are required, so it doesn't really matter what order we do them in.
+
+#### The new handler
+
+Let's create the `App\Handler\AboutPageHandler.php` file.
+So far we have a single module named `App`, so the new Handler goes into the `src/App/src/Handler` folder.
+Go ahead and create the file `src/App/src/Handler/AboutPageHandler.php` with this content:
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Handler;
+
+use Laminas\Diactoros\Response\HtmlResponse;
+use Mezzio\Router\RouterInterface;
+use Mezzio\Template\TemplateRendererInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class AboutPageHandler implements RequestHandlerInterface
+{
+    public function __construct(
+        private string $containerName,
+        private RouterInterface $router,
+        private ?TemplateRendererInterface $template = null
+    ) {
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $data = [];
+        return new HtmlResponse($this->template->render('app::about-page', $data));
+    }
+}
+```
+
+#### The new template
+
+On the last line above there is a template file `app::about-page`.
+If you check `src/App/src/ConfigProvider.php`, you can see what that string means:
+
+- `app` defines a folder in which the template file is kept, in this case `__DIR__ . '/../templates/app'` which resolves to `src/App/templates/app`.
+- `about-page` is the file name without the extensions.
+
+Let's create the About page template named `about-page.html.twig` with a few lines of content for when we check the results of your work:
+
+```twig
+{% extends '@layout/default.html.twig' %}
+
+{% block title %}About Us{% endblock %}
+
+{% block content %}
+    <div class="jumbotron">
+        <h1>About Us</h1>
+        <p>This is some great content!</p>
+    </div>
+{% endblock %}
+```
+
+#### The factory for the new handler
+
+The new handler requires a Factory.
```

Comment:
Auto-wiring should be used via `Laminas\ServiceManager\AbstractFactory\ReflectionBasedAbstractFactory`. Everything else is to complicated.

Reply:
the CLI command mezzio:handler:create still works with Factory 
https://docs.mezzio.dev/mezzio/v3/reference/cli-tooling/